### PR TITLE
fix the wrong TestStart in metamanager_test.go

### PR DESCRIPTION
**Pull request by muxuelan**
**What type of PR is this?**
/kind feature

**What this PR does / why we need it**:
Func TestStart in metamanager_test.go  runs failed due to error: invalid memory address or nil pointer dereference.

**Which issue(s) this PR fixes**:

Fixes #

**Special notes for your reviewer**:

**Does this PR introduce a user-facing change?**:

```release-note

```

### CODE REVIEW

**Comment by daixiang0** [comment]
@muxuelan please dig into this, since all previous CI pass.

**Comment by muxuelan** [comment]
> @muxuelan please dig into this, since all previous CI pass.

I think maybe you can run this test funct locally to verify whether it is correct.

**Comment by muxuelan** [comment]
> @muxuelan please dig into this, since all previous CI pass.



> > @muxuelan please dig into this, since all previous CI pass.
> 
> I think maybe you can run this test funct locally to verify whether it is correct.



> @muxuelan please dig into this, since all previous CI pass.

I think maybe you can run this test func locally to verify whether it is correct

**Comment by muxuelan** [comment]
> @muxuelan please dig into this, since all previous CI pass.

**Comment by muxuelan** [comment]
> @muxuelan please dig into this, since all previous CI pass.

This test is passed, but there is one panic log.

**Comment by daixiang0** [comment]
Please test at local at least 3 times, I check it, panic not always occur.

**Comment by muxuelan** [comment]
> Please test at local at least 3 times, I check it, panic not always occur.
Sorry, the panic occured always at my local, so do my friend.

image

File: edge/pkg/metamanager/metamanager_test.go [hunk 1]
```diff
@@ -59,19 +59,13 @@ func TestNameAndGroup(t *testing.T) {
...
-		t.Errorf("failed to register to beehive")
-	}
+	beehiveContext.InitContext(beehiveContext.MsgCtxTypeChannel)
+	mt := newMetaManager(true)
+	core.Register(mt)
+	beehiveContext.AddModule(mt.Name())
+	beehiveContext.AddModuleGroup(mt.Name(), mt.Group())
 
-	go metaModule.Start()
+	go mt.Start()
```
Comment (fisherxu) [inline edge/pkg/metamanager/metamanager_test.go:68]: Looks good, actually if we move `core.Register(&metaManager{enable: true})` above to the for loop, the panic issue may can also be avoided. @muxuelan  And the InitContext had been called in init func

**Comment by fisherxu** [comment]
@muxuelan Would you like to squash the commits? :)

**Comment by muxuelan** [comment]
> @muxuelan Would you like to squash the commits? :)

I‘v done. ☺

**Comment by fisherxu** [comment]
@muxuelan You can apply for the membership of kubeedge org, xref: https://github.com/kubeedge/community/issues/23

**Comment by muxuelan** [comment]
> @muxuelan You can apply for the membership of kubeedge org, xref: [kubeedge/community#23](https://github.com/kubeedge/community/issues/23)

Wow, it's my pleasure.

**Pull request merged**